sql/core/src/main/scala/org/apache/spark/sql/execution/stat/StatFunctions.scala

```diff
@@ -77,4 +78,42 @@ private[sql] object StatFunctions {
       })
     counts.cov
   }
+
+  /** Generate a table of frequencies for the elements of two columns. */
+  private[sql] def crossTabulate(df: DataFrame, col1: String, col2: String): DataFrame = {
+    val tableName = s"${col1}_$col2"
+    val distinctCol2 = df.select(col2).distinct.collect().sortBy(_.get(0).toString)
```

Comment:
That's what I did first. Xiangrui thought this would be more efficient.

On Fri, May 1, 2015 at 10:16 PM, Reynold Xin <notifications@github.com>
wrote:

> In
> sql/core/src/main/scala/org/apache/spark/sql/execution/stat/StatFunctions.scala
> <https://github.com/apache/spark/pull/5842#discussion_r29545262>:
>
> > @@ -77,4 +78,42 @@ private[sql] object StatFunctions {
> >        })
> >      counts.cov
> >    }
> > +
> > +  /** Generate a table of frequencies for the elements of two columns. */
> > +  private[sql] def crossTabulate(df: DataFrame, col1: String, col2: String): DataFrame = {
> > +    val tableName = s"${col1}_$col2"
> > +    val distinctCol2 = df.select(col2).distinct.collect().sortBy(_.get(0).toString)
>
> btw - isn't a more efficient way to run this is to do groupBy(col1,
> col2).count(), and then pivot the table?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/spark/pull/5842/files#r29545262>.
>